scripts/helpers/iwyu_check.py

```diff
@@ -0,0 +1,194 @@
+    default=False,
+    is_flag=True,
+    help='Skip timestaps in log output')
+@click.option(
+    '--compile-commands-glob',
+    default=os.path.join(proj_root_dir, "out", "debug", "compile_commands*.json"),
+    help='Set global pattern for compile_commands.json files'
+)
+@click.option(
+    '--scanning-destination',
+    default=os.path.join(proj_root_dir, "src", "platform"),
+    help='Set scanning destination file(s) or directory /ies in project'
+)
+@click.option(
+    '--mapping-file-dir',
+    help='Set mapping file directory /ies manually. File should have name iwyu.imp'
+)
+@click.option(
+    '--iwyu-args',
+    default="-Xiwyu --no_fwd_decls -Xiwyu --no_comments",
+    help='Set custom arg(s) for include what you use'
+)
+@click.option(
+    '--clang-args',
+    default="",
+    help='Set custom arg(s) for clang'
+)
+def main(compile_commands_glob, scanning_destination, mapping_file_dir, iwyu_args, clang_args, log_level, no_log_timestamps):
+    # Ensures somewhat pretty logging of what is going on
+    log_fmt = '%(asctime)s %(levelname)-7s %(message)s'
+    if no_log_timestamps:
+        log_fmt = '%(levelname)-7s %(message)s'
+    coloredlogs.install(level=__LOG_LEVELS__[log_level], fmt=log_fmt)
+
+    # checking if a program IWYU exists
+    iwyu = find_program(('iwyu_tool', 'iwyu_tool.py'))
+
+    if iwyu is None:
+        logging.error("Can't find IWYU")
+        sys.exit(1)
+
+    # For iterating how many files had problems with includes
+    warning_in_files = 0
+
+    platform = ""
+    compile_commands_glob = glob.glob(compile_commands_glob)
+
+    if not compile_commands_glob:
+        logging.error("Can't find compile_commands.json file(s)")
+        sys.exit(1)
+
+    for compile_commands in compile_commands_glob:
+
+        compile_commands_path = os.path.dirname(compile_commands)
+        compile_commands_file = os.path.join(compile_commands_path, "compile_commands.json")
+        logging.info("Copy compile command file %s to %s" %
+                     (compile_commands, compile_commands_file))
+
+        with contextlib.suppress(shutil.SameFileError):
+            shutil.copyfile(compile_commands, compile_commands_file)
+
+        # Prase json file for find target name
+        with open(compile_commands, 'r') as json_data:
+            json_data = json.load(json_data)
+
+        for key in json_data:
+            find_re = re.search(r'^.*/src/platform*?\/(.*)/.*$', key['file'])
+            if find_re is not None:
+                platform = find_re.group(1)
+                break
+        if not platform:
+            logging.error("Can't find platform")
+            sys.exit(1)
+
+        if not mapping_file_dir:
+            mapping_file_dir = os.path.join(root_dir + "/platforms/" + platform)
+
+        # Specify config for platforms. Description: Some platform may needed hack clang compiler
+        if platform.lower() == "tizen":
+            if not clang_args:
+                clang_args = [
```

Comment:
I feel (I'm not sure, though) that we should add these options regardless of user input. The "-I" options will do no harm, but it's worth checking whether `--target` can be overwritten if specified latter in the commend line. If it can be overwritten I recommend to use "<built-in options> + <user options>".